app/utils/auth.py

```
@@ -216,7 +219,14 @@ async def get_authenticated_user(
 
     try:
         # Pass the API key directly to the authenticator
-        return await authenticator.authenticate(request, api_key_header)
+        user_claims = await authenticator.authenticate(request, api_key_header)
+
+        # ✨ NEW: Inject git credentials from environment/headers
+        git_creds = extract_git_credentials_from_request(request)
+        user_claims.git_token = git_creds.get("token")
+        user_claims.git_provider = git_creds.get("provider")
+
+        return user_claims
```

Ruff (0.13.2)

Consider moving this statement to an `else` block

(TRY300)